src/user-guide/writing-workflows/scheduler.rst

```diff
@@ -10,53 +10,110 @@ workflows.
 
 
 .. _user_guide.scheduler.workflow_event_handling:
+.. _user_guide.scheduler.workflow_events:
 
-Workflow Event Handling
------------------------
-
-Workflow event handlers allow configured commands to run when workflow events
-occur.
-
-.. note::
+Workflow Events
+---------------
 
-   Cylc supports workflow events e.g. ``startup`` and ``shutdown``
-   and task events e.g. ``submitted`` and ``failed``.
+Cylc supports workflow events which pertain to the :term:`scheduler` e.g.
+``startup`` and ``shutdown`` and task events e.g. ``submitted`` and ``failed``
+which pertain to :term:`tasks <task>`. This section covers workflow events, for
+task events see :ref:`user_guide.runtime.task_event_handling`.
 
-   See also :ref:`user_guide.runtime.task_event_handling`.
+.. rubric:: Event Handlers
 
-Workflow event handlers are configured by:
+Workflow events have "handlers" (i.e. hooks) which allow configured commands to
+run when workflow events occur. These can be configured by:
 
 * :cylc:conf:`flow.cylc[scheduler][events]` (per workflow)
 * :cylc:conf:`global.cylc[scheduler][events]` (user/site defaults)
 
+.. rubric:: Abort On Event
 
-Workflow Events
-^^^^^^^^^^^^^^^
+As well as event handlers, you can tell the scheduler to abort (i.e., shut down
+immediately with error status) on certain workflow events, using the
+``abort on ...`` configurations.
+
+.. rubric:: Configuration
+
+Some workflow events have related configurations e.g. for setting the timeout.
+
+.. rubric:: List of workflow events:
+
+.. cylc-scope:: global.cylc[scheduler][events]
+
+.. describe:: startup
+
+   :Event Handler: `startup handlers`
+
+   The scheduler was started or restarted.
+
+   E.G. using one of these commands ``cylc play``, ``cylc vip`` or ``cylc vr``.
+
+.. describe:: shutdown
+
+   :Event Handler: `shutdown handlers`
+
+   The scheduler was shut down.
+
+   E.G. using the ``cylc stop`` command.
```

Comment:
Doesn't this also apply to aborts and the workflow reaching completion naturally?

Reply:
Natural shutdown yes, not sure about aborts, I would assume not (not a real shutdown, would be odd to get both an aborted and a shutdown event).

Reply:
I have tested with
```
[scheduler]
    [[events]]
        abort on workflow timeout = True
        workflow timeout = PT1S
        abort handlers = echo "ALPHA"
        shutdown handlers = echo "SIERRA"
```

Only the shutdown handler seemed to run
```
ERROR - Workflow shutting down - "abort on workflow timeout" is set
INFO - [('workflow-event-handler-00', 'shutdown') cmd] echo "SIERRA" 'shutdown' 'wflow1' '"abort on workflow timeout" is set'
    [('workflow-event-handler-00', 'shutdown') ret_code] 0
    [('workflow-event-handler-00', 'shutdown') out] SIERRA shutdown wflow1 "abort on workflow timeout" is set
INFO - DONE
```

Reply:
Not sure that's a bug. There's a specific event for workflow timeout.

Reply:
From testing:

* The `abort` handler is only for uncaught internal errors and not for `abort on ...` configs. The language is leading but this behaviour makes sense as we have already fired a more specific event for these cases.
* The `shutdown` handler gets called for pretty much all cases.